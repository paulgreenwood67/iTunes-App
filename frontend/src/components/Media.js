import React from "react";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";



//this component will hold the information about the media to be displayed
export default function Media(props) {
  return (
    <Card
      className="display-card"
      style={{ width: "15rem", marginBottom: "5px" }}
      key={props.trackId}
    >
      <Card.Body>
        <Card.Title>{props.artistName}</Card.Title>
        <Card.Text>{props.trackName}</Card.Text>
        <Card.Text>Genre: {props.primaryGenreName}</Card.Text>
        <Button className="favorites-btn"
          variant="primary"
          onClick={props.addfav}
          data-songid={props.index}
          
        >
        Favourite
        </Button>
      </Card.Body>
    </Card>
  );
}
