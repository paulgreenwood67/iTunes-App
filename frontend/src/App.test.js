import React from "react";
import renderer from "react-test-renderer";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";

// import AppLayout from "../src/components/AppLayout";
// test("renders correctly", () => {
//   const tree = renderer.create(<AppLayout>AppLayout</AppLayout>).toJSON();
//   expect(tree).toMatchSnapshot();
// });

import Media from "../src/components/Media";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <Media>
        <Card>
          <Card.Body>
            <Card.Title></Card.Title>
            <Card.Text></Card.Text>
            <Card.Text></Card.Text>
            <Button></Button>
          </Card.Body>
        </Card>
      </Media>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
