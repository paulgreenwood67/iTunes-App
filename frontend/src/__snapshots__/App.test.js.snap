// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="display-card card"
  style={
    Object {
      "marginBottom": "5px",
      "width": "15rem",
    }
  }
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    />
    <p
      className="card-text"
    >
      Genre: 
    </p>
    <button
      className="favorites-btn btn btn-primary"
      disabled={false}
      type="button"
    >
      Favourite
    </button>
  </div>
</div>
`;
